src/mlsquare/adapters/sklearn.py

```diff
@@ -222,14 +244,17 @@ def predict(self, X):
         1) Write a 'filter_sk_params' function(check keras_regressor wrapper) if necessary.
         2) Data checks and data conversions
         '''
+        if isinstance(self.proxy_model, (sklearn.DimensionalityReductionModel)):
+            raise AttributeError("'SklearnKerasRegressor' object has no attribute 'predict'")
         pred = self.final_model.predict(X)
         return pred
 
     def save(self, filename=None):
         if filename == None:
             raise ValueError(
                 'Name Error: to save the model you need to specify the filename')
-
+        if isinstance(self.proxy_model, (sklearn.DimensionalityReductionModel)):
```

Comment:
why are still you checking for these kind of isinstance of kind of checks? try avoiding.